src/arithmetic.jl

```diff
@@ -405,6 +405,18 @@ for (f, fc) in ((:+, :(add!)), (:-, :(subst!)))
     end
 end
 
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        function sum!(v::$T{S}, a::AbstractArray{$T{S}}) where {S <: Number}
```

Comment:
Outside from this for-loop, I suggest to add a method for `sum!` with the signature`sum!(v::TaylorN{S}, a::AbstractArray{HomogeneousPolynomial{S}})`, which is used in some `evaluate` and/or related functions.

Reply:
Thank you for the suggestion! I've added the corresponding method, although I don't know which method of `evaluate` will benefit the most from it.